SWIPs/swip-20-honey_to_money.md

```diff
@@ -0,0 +1,103 @@
+- Zero marginal costs of serving data with perfect competition means that when nodes would engage in peer-to-peer price discovery, the price will tend to zero and a sub-optimal equilibrium of near-zero prices will be reached.
+- The high-quantity of transactions implies that nodes cannot engage in price-discovery for every byte of data transacted, meaning they have to agree *beforehand* on how they will come to a price *at a later time*.
+- Nodes are most of the time just conduits for data and payments, meaning that most of the time data requests (and thus also payments) will take place because nodes want to send data to a peer who requested this data from them.
+
+### Discussion on time differences causing imbalances
+#### Delay between sending and receiving a payment
+Nodes might apply different honey prices for the same payment, due to a time difference between applying a price for *sending* a payment and applying a price for *receiving* the same payment. The imbalances caused by this don't do significant harm if the sum of all imbalances is not greater than the disconnect threshold. Theoretically, if the overall price movement does not show a clear trend (either up or down) over the period where nodes do business with each other, or the two nodes are neither a net receiver nor a net provider to each other, accounting imbalances should offset each other. Furthermore, less frequent as well as small price updates should minimize the imbalances. If a clear trend exists and one node is a net receiver and the other a net provider, the disconnect threshold will eventually be reached. Research must be performed on how the strength of a trend influences the duration until a disconnect is reached for a given stream of traffic and whether this is expected to cause harm to the connectivity in Swarm. The assumption is that this will not be the case for a typical Swarm node. 
+
+If there appears to be a problem with such imbalances, we see a possible solution: as nobody benefits from price imbalances, both the sender and the receiver of a payment are incentivized to apply a price which is not per their clock, but per the clock of the counterparty when they expect the counterparty to query or have queried the oracle. To assist in this, a payload might be sent with the transaction, specifying the clock with which the oracle price was applied.
```

Comment:
> as nobody benefits from price imbalances

Why is this the case?

If I was the providing node and benefited from higher prices, can't I just adopt the attitude that nodes simply top up their credit to afford consumption?

Reply:
> Why is this the case?

A price imbalance comes into existence if one node (A) applies a higher price than the other (B). Sure, if A is the net payer, B will benefit, but here it is meant that B is the net user, hence the price balance means that A get's less than expected, while B pays exactly what he thinks he should pay. This becomes a problem when this price balance becomes sufficiently big to lead to a disconnect and nobody benefits from disconnects.

I would rephrase it as: `as nobody benefits from price imbalances that lead to disconnects, ...`